Foundation/Conduit/Textual.hs

```diff
@@ -0,0 +1,46 @@
+module Foundation.Conduit.Textual
+    ( lines
+    , fromBytes
+    , toBytes
+    ) where
+
+import           Foundation.Internal.Base hiding (throw)
+import           Foundation.Array.Unboxed
+import           Foundation.String (String)
+import qualified Foundation.String.UTF8 as S
+import           Foundation.Conduit.Internal
+import           Foundation.Monad
+
+-- | Split conduit of string to its lines
+--
+-- This is very similar to Prelude lines except
+-- it work directly on Conduit
+lines :: Monad m => Conduit String String m ()
+lines = await >>= maybe (finish mempty) (go mempty)
+  where
+    finish buf = if S.null buf then return () else yield buf
+
+    go current nextBuf =
+        case S.uncons next' of
+            Just (_, rest') -> yield (current `mappend` line) >> go mempty rest'
+            Nothing         ->
+                let nextCurrent = current `mappend` nextBuf
```

Comment:
yeah good call; It did morally hurt me to actually merge the string this way, but had to be done in the spirit of getting something right now without more yak shaving. I'll seek to fix it before merging, but sadly the string builder is still pretty much only in my head